src/interfaces/records/handlers/records-delete.ts

```diff
@@ -58,6 +58,8 @@ export class RecordsDeleteHandler implements MethodHandler {
       newestMessage = newestExistingMessage;
     }
 
+    await deleteAllOlderData(tenant, existingMessages, newestMessage, this.dataStore);
```

Comment:
Can we move `deleteAllOlderData()` call after `deleteAllOlderMessagesButKeepInitialWrite()` call below (as the last call).

Reason being mainly in case `put()` blows up, the record wouldn't be in a bad state, and secondarily it's nice that data state follows the message state.